homeassistant/components/huum/climate.py

```diff
@@ -0,0 +1,135 @@
+"""Support for Huum wifi-enabled sauna."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from huum.const import SaunaStatus
+from huum.exceptions import SafetyException
+from huum.huum import Huum
+from huum.schemas import HuumStatusResponse
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import ClimateEntityFeature, HVACMode
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Huum sauna with config flow."""
+    huum_handler = Huum(
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        entry_status = await huum_handler.status_from_status_or_stop()
+    except Exception as status_error:
+        raise PlatformNotReady(status_error) from status_error
+
+    async_add_entities([HuumDevice(entry_status, huum_handler)])
+
+
+class HuumDevice(ClimateEntity):
+    """Representation of a heater."""
+
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_max_temp = 110
+    _attr_min_temp = 40
+
+    _target_temperature = 40
+
+    def __init__(self, status: HuumStatusResponse, huum_handler: Huum) -> None:
+        """Initialize the heater."""
+        self._status = status
+        self._huum_handler = huum_handler
+
+    @property
+    def name(self) -> str:
```

Comment:
By "adding a DeviceInfo" do you refer to this documentation https://developers.home-assistant.io/docs/device_registry_index?_highlight=deviceinfo#defining-devices

So I would add a `device_info` property function that returns a `DeviceInfo`?

Reply:
or `_attr_device_info` in the constructor, which might be cleaner to do

Reply:
But yes

Reply:
In this case, I can't get a unique identifier of the sauna entity. Is it OK to just hard-code the name "Huum Sauna" as the name, so that the settings would be something in the line of:
```python
    self._attr_device_info = DeviceInfo(
        name="Huum Sauna",
        manufacturer="Huum",
    )
```

Reply:
Please then set the config_entry.entry_id as unique_id for both the entity and the device

Reply:
Sorry, this is my first integration, and there is quite a bit to learn here 😄 

How do I go about doing that? 
Is something like this what you are looking for?
``` python
class HuumDevice(ClimateEntity):
    self._attr_device_info = DeviceInfo(
        name="Huum Sauna",
        manufacturer="Huum",
    )
    def __init__(self, status: HuumStatusResponse, huum_handler: Huum) -> None:
        self.unique_id  = self.entity_id
```

Reply:
1. Add `unique_id` as parameter to HuumDevice
2. in `async_setup_entry` fill it with `config_entry.entry_id`.


```py
def __init__(self, huum_handler: Huum, unique_id: str) -> None:
    self._attr_unique_id = unique_id
    self._attr_device_info = DeviceInfo(
        identifiers={(DOMAIN, unique_id)},
        name="Huum sauna",
    )
```

Reply:
Thank you for all the help. Changed the code accordingly now.